### PR TITLE
Add code transforms for Atomist peer dependencies and package.json author

### PR DESCRIPTION
**Pull request by lievendoclo**

### CODE REVIEW

File: src/machine/tryToUpdateAtomistPeerDependencies.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+@Parameters()
+class UpdateAtomistPeerDependenciesParameters {
+    public commitMessage: string;
+
+}
+
+const UpdateAtomistPeerDependenciesStarTransform: CodeTransform<UpdateAtomistPeerDependenciesParameters> =
+    async (p, ctx, params) => {
+        const pjFile = await p.getFile("package.json");
+        const pj = JSON.parse(await pjFile.getContent());
```
Comment (ddgenome) [inline src/machine/tryToUpdateAtomistPeerDependencies.ts]: Always good to use a try/catch block around I/O, JSON parsing, etc. Maybe just throw a try/catch around the whole function.

File: src/machine/tryToUpdateAtomistPeerDependencies.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+async function updatePeerDependencies(deps: any,
+                                      versions: string[],
+                                      sendMessage: (msg?: string) => Promise<void>): Promise<void> {
+    for (const k in deps) {
+        if (deps.hasOwnProperty(k)) {
+            if (k.startsWith("@atomist/")) {
+                const oldVersion = deps[k];
+                const version = `*`;
+                if (version && oldVersion !== version) {
```
Comment (ddgenome) [inline src/machine/tryToUpdateAtomistPeerDependencies.ts]: Why are you testing the truthiness of `version`? You set it on the line above. Did you mean to test `oldVersion`?

File: src/machine/tryToUpdateAtomistPeerDependencies.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+class UpdateAtomistPeerDependenciesParameters {
+    public commitMessage: string;
+
+}
+
+const UpdateAtomistPeerDependenciesStarTransform: CodeTransform<UpdateAtomistPeerDependenciesParameters> =
+    async (p, ctx, params) => {
+        const pjFile = await p.getFile("package.json");
+        const pj = JSON.parse(await pjFile.getContent());
+        const versions = [];
```
Comment (ddgenome) [inline src/machine/tryToUpdateAtomistPeerDependencies.ts]: This is implicitly an array of `any`. It's an array of strings right?

```typescript
const versions: string[] = [];
```

File: src/machine/tryToUpdateAtomistPeerDependencies.ts [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+            if (msg) {
+                message.attachments[0].text = `${message.attachments[0].text}${msg}`;
+            }
+            await ctx.messageClient.respond(message, opts);
+        };
+
+        await sendMessage();
+
+        if (pj.peerDependencies) {
+            await updatePeerDependencies(pj.peerDependencies, versions, sendMessage);
```
Comment (ddgenome) [inline src/machine/tryToUpdateAtomistPeerDependencies.ts]: Why pass in the empty array to have a function which returns nothing populate? It's much better to have the function return the array of strings.

File: src/machine/tryToUpdateAtomistPeerDependencies.ts [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+                                      versions: string[],
+                                      sendMessage: (msg?: string) => Promise<void>): Promise<void> {
+    for (const k in deps) {
+        if (deps.hasOwnProperty(k)) {
+            if (k.startsWith("@atomist/")) {
+                const oldVersion = deps[k];
+                const version = `*`;
+                if (version && oldVersion !== version) {
+                    deps[k] = version;
+                    versions.push(`${k} ${oldVersion} > ${version}`);
```
Comment (ddgenome) [inline src/machine/tryToUpdateAtomistPeerDependencies.ts]: Again, it seems odd that this array is used to pass information back to the caller. It should just be returned.

File: src/machine/nodeSupport.ts [hunk 6]
```diff
@@ -246,7 +248,9 @@ export function addNodeSupport(sdm: SoftwareDeliveryMachine): SoftwareDeliveryMa
         sdm.configuration.sdm.npm as NpmOptions));
 
     sdm.addCodeTransformCommand(TryToUpdateAtomistDependencies)
-        .addCodeTransformCommand(UpdatePackageVersion);
+        .addCodeTransformCommand(UpdatePackageVersion)
+        .addCodeTransformCommand(TryToUpdateAtomistPeerDependencies)
+        .addCodeTransformCommand(UpdatePackageAuthor);
```
Comment (ddgenome) [inline src/machine/nodeSupport.ts]: Perhaps these should be autofixes?

Comment (lievendoclo) [reply src/machine/nodeSupport.ts]: We decided to go with a command first

File: src/machine/tryToUpdateAtomistPeerDependencies.ts [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+
+            params.commitMessage = `Update @atomist NPM peer dependencies to *
+
+${versions.join("\n")}
+
+${DryRunMessage} ${AutoMergeCheckSuccessTag}`;
+
+            return p;
+        } catch (e) {
+            await ctx.messageClient.respond(`\n:atomist_build_failed: Updating peer dependencies in package.json failed`);
```
Comment (ddgenome) [inline src/machine/tryToUpdateAtomistPeerDependencies.ts]: It's a good idea to include `e.message` in the response, or at least log it.

**Pull request merged**